client/src/components/SearchBar.vue

```diff
@@ -0,0 +1,102 @@
+<template>
+  <div class='container'>
+    <div class='data-layer-options'>
+      <search-bar-option
+        v-for='option in options'
+        :key='option'
+        :text-content='option.name'
+        :selected='getSelected(option)'
+        @click='setSelected(option)' />
+    </div>
+    <div class='right-align'>
+      <div class='select-wrapper'>
+        <vue-select
+          label='name'
+          v-model='this.selectedOption'
+          :options='this.options' />
+      </div>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import { defineComponent, inject } from 'vue';
+import SearchBarOption from './SearchBarOption.vue';
+import VueSelect from 'vue-select';
+import 'vue-select/dist/vue-select.css';
+import { State } from '../model/state';
+import { stateKey } from '../injection_keys';
+import { DataLayer } from '../model/data_layer';
+
+export default defineComponent({
+  name: 'SearchBar',
+  components: { SearchBarOption, VueSelect },
+  setup() {
+    const state: State | undefined = inject(stateKey);
+
+    return {
+      state,
+    };
+  },
+  data() {
+    return {
+      options: [] as DataLayer[],
+      selectedOption: null as DataLayer | null,
```

Comment:
I'm not 100% sure but can you do "null as DataLayer?" ?

Reply:
Required some other funky thing, but left a comment with explanation. Lmk if this doesn't make sense.